src/utils/deviceInfo/deepDeviceInfo.ts

```diff
@@ -0,0 +1,172 @@
+import UAParser from 'ua-parser-js';
+import { DetailedDeviceInformation, SupportedPlatforms } from '../../types/device';
+
+interface DetailedDeviceInformationParserProps {
+    deviceInformation: DetailedDeviceInformation;
+    uaDataValues?: UADataValues;
+    uaParser: UAParser;
+}
+
+/**
+ * This object is concerned with gathering and holding information about the users device.
+ * This object will be read-only after initialization and should as a rule throw errors if the
+ * deviceInformation property is altered after intialization.
+ */
+class DetailedDeviceInformationProvider {
```

Comment:
Any chance to add unit tests for this, and the other classes? I see that in the original PR the tests were missing as well.

Reply:
The tests are in place in this PR

Reply:
My bad: it seems I am blind, sorry for that 🙈